helm/hatchery/templates/workspace-launch-test-cron.yaml

```diff
@@ -0,0 +1,231 @@
+                    awk -F . '{ print $2 }' /mnt/shared/access_token.txt | base64 --decode 2> /dev/null
+                    success=true
+                  else
+                    echo "fence-create failed!"
+                    cat "$tempFile"
+                    echo "sleep for $sleepTime, then retry"
+                    sleep "$sleepTime"
+                    let sleepTime=$sleepTime+$sleepTime
+                  fi
+                  let count=$count+1
+                done
+                if [[ $success != true ]]; then
+                  echo "Giving up on fence-create after $count retries - failed to create valid access token"
+                fi
+                echo ""
+                echo "All Done - always succeed to avoid k8s retries"
+          - name: workspace-launch-test
+            env:
+              - name: GEN3_HOSTNAME
+                value: {{ .Values.workspaceLaunchTest.hostname | quote}}
+              - name: WORKSPACE_LAUNCH_IMAGES
+                value: {{ .Values.workspaceLaunchTest.workspaceImages | default "(Generic) Jupyter Lab Image with R Kernel" | quote }}
+            image: quay.io/cdis/awshelper:master
+            imagePullPolicy: Always
+            volumeMounts:
+              - name: shared-data
+                mountPath: /mnt/shared
+            resources:
+              limits:
+                cpu: 3
+                memory: "512Mi"
+              requests:
+                cpu: "1"
+                memory: "256Mi"
+            command: ["/bin/bash"]
+            args:
+              - "-c"
+              - |
+                if [[ -z "$WORKSPACE_LAUNCH_IMAGES" ]]; then
+                  if [[ ! -z "$WORKSPACE_LAUNCH_IMAGES" ]]; then
+                    echo "WORKSPACE_LAUNCH_IMAGES is from g3auto secret"
+                    export WORKSPACE_LAUNCH_IMAGES=$WORKSPACE_LAUNCH_IMAGES
+                  else
+                    echo -e "ERROR: WORKSPACE_LAUNCH_IMAGES must be in secret or on command line" 1>&2
+                    exit 0
+                  fi
+                else
+                  echo "WORKSPACE_LAUNCH_IMAGES is from command line parameter"
+                fi
+
+                # wait for the access token to be created in the sidecar "fence" container
+                let count = 0
+                while [[ ! -f /mnt/shared/access_token.txt && $count -lt 50 ]]; do
+                  echo "waiting for /mnt/shared/access_token.txt";
+                  sleep 2
+                  let count=$count+1
+                done
+
+                export ACCESS_TOKEN="$(cat /mnt/shared/access_token.txt)"
+
+                echo "Installing dependencies..."
+                python3 -m pip3 install requests
+                python3 -m pip3 install logging
+
+                ls
+
+                cmd="python3 cloud-automation/files/scripts/workspaces_launch_test.py --commons-url $GEN3_HOSTNAME --access-token $ACCESS_TOKEN"
```

Comment:
Should this be using cloud-automation? 🤔

Reply:
I asked Ed about it earlier. I can't get any files from gen3-helm into this job. But since its using the aws-helper image, im able to run this script.

Reply:
😬 But does that image get the file from the scripts folder here? I think we're creating too much confusion by including the script in helm but then not using it and relying on a separate image where the script is still maintained in cloud automation. I'm not sure we should allow using old cloud auto code in the aws-helper image 🤷 open to arguments, but we're coupling helm with cloud auto here if we do this. I would argue we should not do that.

I might need more context to provide more useful feedback, but maybe we just embed the whole python script into a command here instead of having it in a separate file at all?